jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java

```diff
@@ -395,6 +395,39 @@ public float[] getBaseMargin() throws XGBoostError {
     return getFloatInfo("base_margin");
   }
 
+  /**
+   * Get feature data as BigDenseMatrix
+   * @return feature Matrix
+   * @throws XGBoostError
+   */
+  public BigDenseMatrix getData() throws XGBoostError {
+    int rowNum = (int) rowNum();
+    int colNum = (int) colNum();
+    int nonMissingNum = (int) nonMissingNum();
+
+    long[] rowOffset = new long[rowNum + 1];
+    int[] featureIndex = new int[nonMissingNum];
+    float[] featureValue = new float[nonMissingNum];
+
+    XGBoostJNI.checkCall(XGBoostJNI.XGDMatrixGetDataAsCSR(handle, "{}",
+        rowOffset, featureIndex, featureValue));
+
+    BigDenseMatrix denseMatrix = new BigDenseMatrix(rowNum, colNum);
+
+    for (int row = 0; row < rowNum; row++) {
+      for(int col = 0; col < colNum; col++){
+        denseMatrix.set(row, col, 0.0f);
```

Comment:
is it possible to set the default to 0.0f when creating BigDenseMatrix

Reply:
sorry for the late response. No such functions provided by BigDenseMatrix.

Reply:
Can we just return the CSR instead? Converting a CSR to a dense can be extremely memory hungry when input is sparse. Some datasets can span a few thousands of features with only a small portion of valid values due to encoding.